docs/en/operation/Troubleshooting.md

```diff
@@ -269,6 +269,71 @@ are
 - `ADVANCED`: samples and tracks recent accesses, higher overhead
 - `PARANOID`: tracks for leaks on every resource allocation, highest overhead. 
 
+## Master Internal Monitoring
+
+Alluxio master periodically checks its resource usage, including CPU and memory usage, and several internal data structures 
+that are performance critical. This interval is configured by `alluxio.master.throttle.heartbeat.interval` (defaults to 3 seconds).
+On every sampling point in time (PIT), Alluxio master takes a snapshot of its resource usage. A continuous number of PIT snapshots
+(number configured by alluxio.master.throttle.observed.pit.number, defaults to 3) will be saved and used to generate the aggregated
+resource usage which is used to decide the system status.
+
+Each PIT includes the following metrics.
+
+```console
+directMemUsed=5268082, heapMax=59846950912, heapUsed=53165684872, cpuLoad=0.4453061982287778, pitTotalJVMPauseTimeMS=190107, totalJVMPauseTimeMS=0, rpcQueueSize=0, pitTimeMS=1665995384998}
+```
+
+- `directMemUsed`: direct memory allocated by `ByteBuffer.allocateDirect`
+- `heapMax` : the allowed max heap size
+- `heapUsed` : the heap memory used
+- `cpuLoad` : the cpu load
+- `pitTotalJVMPauseTimeMS` : aggregated total JVM pause time from the beginning
+- `totalJVMPauseTimeMS` : the JVM pause time since last PIT
+- `rpcQueueSize` : the rpc queue size
+- `pitTimeMS` : the timestamp in millisecond when this snapshot is taken
+
+The aggregated server indicators are the certain number of continuous PITs, this one is generated in a sliding window. The alluxio
+master has a derived indicator `Master.system.status` that is based on the heuristic algorithm.
+
+```console
+    "Master.system.status" : {
+      "value" : "STRESSED"
+    }
+```
+
+The possible statuses are: 
+- `IDLE`
+- `ACTIVE`
+- `STRESSED`
+- `OVERLOADED`
+
+The system.status is mainly decided by the JVM pause time and the free heap memory. Usually the status transition is 
+`IDLE` <---> `ACTIVE` <---> `STRESSED` <---> `OVERLOADED`
+- If the JVM pause time is longer than `alluxio.master.throttle.overloaded.heap.gc.time`, the system.status is directly set to `OVERLOADED`.
+- If the used heap memory is greater than the low used heap memory boundary threshold, the system.status is deescalated.
+- If the used heap memory is less than the upper used heap memory boundary threshold, the system.status is unchanged.
+- If the aggregated used heap memory is greater than the upper used heap memory boundary threshold, the sytem.status is escalated.
+- As the used heap memory grows or shrinks, the value of the system status will update if it crosses any of the thresholds defined by the configurations below
```

Comment:
Are these conditions evaluated in order as described here? Is so, I suggest using an ordered list, instead of an unordered one.

You can do so by:

```
1. A
2. B
3. C

```

If not, then I think we need more clarification on how exactly are the rules. But again, I think that would be too much implementation details.